django/db/models/query_utils.py

```diff
@@ -99,6 +99,45 @@ def resolve_expression(
         query.promote_joins(joins)
         return clause
 
+    def replace_expressions(self, replacements):
+        if not replacements:
+            return self
+        clone = self.create(connector=self.connector, negated=self.negated)
+        for child in self.children:
+            child_replacement = child
+            if isinstance(child, tuple):
+                lhs, rhs = child
+                if LOOKUP_SEP in lhs:
+                    path, lookup = lhs.rsplit(LOOKUP_SEP, 1)
+                else:
+                    path = lhs
+                    lookup = None
+                field = models.F(path)
+                if (
+                    field_replacement := field.replace_expressions(replacements)
+                ) is not field or lookup is None:
```

Comment:
```suggestion
                ) is not field:
```
I can remove `or lookup is None` without test failures, do we need to keep it?

Reply:
You're right, I spent a bit of time moving things around and reason about it and I believe it's an artifact of a previous implementation.